transport/src/main/java/io/netty/bootstrap/AbstractBootstrap.java

```diff
@@ -375,31 +376,66 @@ public B handler(ChannelHandler handler) {
         return (B) this;
     }
 
-    final SocketAddress localAddress() {
+    /**
+     * Return the configured local address or {@code null} if non is configured yet.
+     */
+    public final SocketAddress localAddress() {
         return localAddress;
     }
 
+    /**
+     * Return the configured {@link ChannelFactory} or {@code null} if non is configured yet.
+     * Be aware if you call {@link #channel(Class)} it will create the {@link ChannelFactory} for you.
+     */
     @SuppressWarnings("deprecation")
-    final ChannelFactory<? extends C> channelFactory() {
+    public final ChannelFactory<? extends C> channelFactory() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove usage of generic wildcard type. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1452)